lib/cfn-nag/custom_rules/ApiGatewayMethodAuthorizationTypeRule.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/violation'
+require_relative 'base'
+
+class ApiGatewayMethodAuthorizationTypeRule < BaseRule
+  def rule_text
+    "AWS::ApiGateway::Method should not have AuthorizationType set to 'NONE'. "
+  end
+
+  def rule_type
+    Violation::WARNING
+  end
+
+  def rule_id
+    'W58'
+  end
+
+  def audit_impl(cfn_model)
+    violating_deployments = cfn_model.resources_by_type('AWS::ApiGateway::Method').select do |method|
+      method.authorizationType.nil? || method.authorizationType.to_s.downcase == 'none'
```

Comment:
technically nil.to_s == ''

Reply:
to_s.downcase is cool, but I believe we've got a bunch of prior art/style around using strcmpi

Reply:
@erickascic  Just pushed up some slight changes. Is that what you had in mind?

Reply:
ok i made a half stupid comment.  i like your original .nil? call better.  i was thinking nil being empty would fall through properly in the casecmp but it won't.  sorry for the runaround.